packages/x-date-pickers/src/DesktopDateTimePicker/DesktopDateTimePicker.tsx

```diff
@@ -28,24 +34,44 @@ const DesktopDateTimePicker = React.forwardRef(function DesktopDateTimePicker<TD
     TDate,
     DesktopDateTimePickerProps<TDate>
   >(inProps, 'MuiDesktopDateTimePicker');
+  const thresholdToRenderTimeInASingleColumn =
+    defaultizedProps.thresholdToRenderTimeInASingleColumn ?? 24;
+  const timeStep = defaultizedProps.timeStep ?? 5;
+  const shouldRenderTimeInASingleColumn =
+    (24 * 60) / timeStep <= thresholdToRenderTimeInASingleColumn;
+
+  const renderTimeView = shouldRenderTimeInASingleColumn
+    ? renderDigitalClockTimeView
+    : renderMultiSectionDigitalClockTimeView;
 
   const viewRenderers: PickerViewRendererLookup<TDate | null, DateOrTimeView, any, {}> = {
     day: renderDateViewCalendar,
     month: renderDateViewCalendar,
     year: renderDateViewCalendar,
-    hours: null,
-    minutes: null,
-    seconds: null,
+    hours: renderTimeView,
+    minutes: renderTimeView,
+    seconds: renderTimeView,
     ...defaultizedProps.viewRenderers,
   };
   const ampmInClock = defaultizedProps.ampmInClock ?? true;
+  const actionBarActions: PickersActionBarAction[] = !shouldRenderTimeInASingleColumn
+    ? ['accept']
+    : [];
 
   // Props with the default values specific to the desktop variant
   const props = {
     ...defaultizedProps,
     viewRenderers,
     yearsPerRow: defaultizedProps.yearsPerRow ?? 4,
     ampmInClock,
+    timeStep,
+    // fallback to `closeOnSelect={false}` given a multi column time picker.
+    closeOnSelect: inProps.closeOnSelect ?? shouldRenderTimeInASingleColumn,
+    // Setting single time view in case of single column time picker
+    // Allows for easy view lifecycle management
+    views: shouldRenderTimeInASingleColumn
+      ? defaultizedProps.views.filter((view) => !isTimeView(view) || view === 'hours')
```

Comment:
Does it work with `<DateTimePicker views={['date', 'minutes', 'seconds']} />` ? :angel: 

(yes that's a very weird use case)

Reply:
Nope, it clearly won't work in such case.
How would you expect the component to behave?
- Forcefully add `hours` view (regardless if it is present) and function in single column mode if other conditions call for it (`timeStep` & `thresholdToRenderTimeInASingleColumn`)?
- Fallback to rendering the multi column view if provided `views` do not allow for single-column?

I tend to lean towards 1st option, something like:
```ts
views: shouldRenderTimeInASingleColumn
    ? new Set([...defaultizedProps.views.filter((view) => !isTimeView(view) || view === 'hours'), 'hours']),
    : defaultizedProps.views,
```

Reply:
This is currently no longer relevant as we do not plan to ship DigitalClock on `DateTimePicker`. 👌